pkg/generator/certificate.go

```diff
@@ -156,32 +156,32 @@ func (kp *CertKeyPair) References() ([]string, []string) {
 }
 
 // LoadSecretFromManager populates RootCA data from secret manager
-func (kp *CertKeyPair) LoadSecretFromManager(ctx context.Context, config *v1alpha1.AppConfig, namespace, secretName string) error {
+func (kp *CertKeyPair) LoadSecretFromManager(ctx context.Context, sm *secretsmanager.SecretManager, namespace, secretName string) error {
 	var err error
 	publicPemKeyFmt := fmt.Sprintf("%s_%s_%s.pem", namespace, secretName, kp.Name)
 	privatePemKeyFmt := fmt.Sprintf("%s_%s_%s-private.pem", namespace, secretName, kp.Name)
 
-	kp.Cert.CertPEM, err = secretsmanager.LoadSecret(ctx, config, publicPemKeyFmt)
+	kp.Cert.CertPEM, err = (*sm).LoadSecret(ctx, publicPemKeyFmt)
```

Comment:
why are these calls being casted down? also wouldn't need to de reference... with my earlier comment.

Reply:
Updated along with previous commit

Reply:
you still have the type casting `(sm).LoadSecret` that shouldn't be needed.

Reply:
fixed